estate/models/estate_property.py

```diff
@@ -0,0 +1,153 @@
+from odoo import fields, models, api
+from odoo.exceptions import UserError
+from datetime import date
+from dateutil.relativedelta import relativedelta
+from odoo.tools.float_utils import float_compare, float_is_zero
+from odoo.exceptions import ValidationError
+
+
+class EstateProperty(models.Model):
+    _name = "estate.property"
+    _description = "Real Estate Property"
+
+    _sql_constraints = [
+        (
+            "expected_price_strictly_positive",
+            "CHECK(expected_price > 0)",
+            "Expected price must be strictly positive",
+        ),
+        (
+            "selling_price_positive",
+            "CHECK(selling_price >= 0)",
+            "Selling price must be positive",
+        ),
+    ]
+
+    @api.constrains("expected_price", "selling_price")
+    def _check_selling_price(self):
+        for record in self:
+            if not float_is_zero(record.selling_price, precision_digits=2):
+                if (
+                    float_compare(
+                        record.selling_price,
+                        record.expected_price * 0.9,
+                        precision_digits=2,
+                    )
+                    < 0
+                ):
+                    raise ValidationError(
+                        "The selling price cannot be lower than 90% of the expected price!"
+                    )
+
+    def action_sold(self):
+        for record in self:
+            if record.state != "cancelled":
+                record.state = "sold"
+                return True
+            raise UserError("A cancelled property cannot be set as sold")
```

Comment:
```suggestion
            raise UserError(_("A cancelled property cannot be set as sold"))
```
Some strings are automatically translated but this is not the case for error messages. To make it translatable, use `_()` method :+1: